integrations-and-supported-tools/pytorch/scripts/Neptune_Pytorch_Support_more_options.py

```diff
@@ -0,0 +1,161 @@
+from neptune.new.types import File
+import torch.nn.functional as F
+
+
+# Step 1: Initialize Neptune and create new Neptune Run
+run = neptune.init(project="common/pytorch-integration", tags='More options script', api_token="ANONYMOUS", source_files=["*.py"])
+
+
+# Experiment Config
+data_dir = "data/CIFAR10"
+data_tfms = {
+    "train": transforms.Compose([
+        transforms.RandomHorizontalFlip(),
+        transforms.ToTensor(),
+        transforms.Normalize([0.485, 0.456, 0.406], [0.229, 0.224, 0.225])
+    ]),
+    "val": transforms.Compose([
+        transforms.ToTensor(),
+        transforms.Normalize([0.485, 0.456, 0.406], [0.229, 0.224, 0.225])
+        
+    ])
+}
+
+params = {
+    "lr": 1e-2,
+    "bs": 128,
+    "input_sz": 32 * 32 * 3,
+    "n_classes": 10,
+    "epochs": 1,
+    "model_filename": "basemodel",
+    "device": torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+}
+
+# Model & Dataset
+class BaseModel(nn.Module):
+    def __init__(self, input_sz, hidden_dim, n_classes):
+        super(BaseModel, self).__init__()
+        self.main = nn.Sequential(
+            nn.Linear(input_sz, hidden_dim*2),
+            nn.ReLU(),
+            nn.Linear(hidden_dim*2, hidden_dim),
+            nn.ReLU(),
+            nn.Linear(hidden_dim, hidden_dim//2),
+            nn.ReLU(),
+            nn.Linear(hidden_dim//2, n_classes)
+        )
+
+    def forward(self, input):
+        x = input.view(-1, 32* 32 * 3)
+        return self.main(x)
+
+trainset = datasets.CIFAR10(data_dir, transform=data_tfms["train"], 
+                            download=True)
+trainloader = torch.utils.data.DataLoader(trainset, 
+                                        batch_size=params["bs"],
+                                        shuffle=True, num_workers=2)
+
+validset = datasets.CIFAR10(data_dir, train=False,
+                        transform=data_tfms["train"],
+                        download=True)
+validloader = torch.utils.data.DataLoader(validset, 
+                                        batch_size=params["bs"], 
+                                        num_workers=2)
+dataset_size = {"train": len(trainset), "val": len(validset)}
+
+# Instatiate model, criterion and optimizer
+model = BaseModel(params["input_sz"], params["input_sz"], params["n_classes"]).to(params["device"])
+criterion = nn.CrossEntropyLoss()
+optimizer = optim.SGD(model.parameters(), lr=params["lr"])
+
+classes = ["airplane", "automobile", "bird", "cat", "deer", "dog", "frog","horse","ship","truck"]
+
+
+# Step 2: Log config & hyperpararameters
+run["config/dataset/path"] = data_dir
+run["config/dataset/transforms"] = data_tfms
+run["config/dataset/size"] = dataset_size
+run["config/model"] = type(model).__name__
+run["config/criterion"] = type(criterion).__name__
+run["config/optimizer"] = type(optimizer).__name__
+run["config/hyperparameters"] = params
+run["config/classes"] = classes
+
+# Step 3: Log losses and metrics 
+for epoch in range(params["epochs"]):
```

Comment:
I think we can drop this loop and epoch params

Reply:
Really?